### ISSUE TITLE
Fix TS bugs as a result of upgrading @wordpress packages to wp-6.2

### ISSUE THREAD

**Issue by samueljseay** [woothemes/woocommerce#39118]
Upgrading the `@wordpress` packages across the repository to wp-6.2 exposes a number of TS build errors.

These errors need to be fixed before we can ship the update. Rather than creating many individual issues this issue will track a task list of things we need to fix. I will put individual error reports in comments.

There is a draft PR here where fixes can be pushed: https://github.com/woocommerce/woocommerce/pull/39117

IMPORTANT EXTRA INFO FOR WORKING ON THIS TASK:

if you run `pnpm build` from any directory it's going to try build dependant packages first before building yours, thereby you won't see package errors because the deps fail first. To get around this in your environment you need to still build the package deps but have them not emit an error. This is simple to do, in any dependant package that fails it will have a script something like:

```
"turbo:build": "tsc --project tsconfig.json && tsc --project tsconfig-cjs.json"
```

during testing and working on fixes replace it with:

```
"turbo:build": "(tsc --project tsconfig.json || echo 'TS errors but continue')"
```
This will still emit everything but it won't signal failure so you can see the failures of your individual package.

```[tasklist]
### Tasks
- [x] Remove redeclarations of `@wordpress/data`
- [x] Error in `packages/js/notices`: `src/store/index.ts(20,16): error TS2347: Untyped function calls may not accept type arguments.`
- [x] Fix errors in `packages/js/data` see https://github.com/woocommerce/woocommerce/issues/39118#issuecomment-1623045302
- [x] Fix errors in `packages/js/components` see https://github.com/woocommerce/woocommerce/issues/39118#issuecomment-1623056592
- [x] Fix errors in `packages/js/customer-effort-score`: https://github.com/woocommerce/woocommerce/issues/39118#issuecomment-1623059411
- [x] Fix errors in `packages/js/product-editor`: https://github.com/woocommerce/woocommerce/issues/39118#issuecomment-1623059411
- [ ] Fix errors in `plugins/woocommerce-admin` (The error report is too long to paste because it pastes error locations in code)
```

**Comment by samueljseay** [comment]
Error report for `packages/js/data`:

```
@woocommerce/data:turbo:build: src/countries/index.ts(21,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/crud/index.ts(4,42): error TS2305: Module '"@wordpress/data"' has no exported member 'StoreConfig'.
@woocommerce/data:turbo:build: src/export/index.ts(20,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/import/index.ts(21,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/items/index.ts(22,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/items/utils.ts(66,23): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/navigation/index.ts(21,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/navigation/with-navigation-hydration.tsx(21,30): error TS2558: Expected 2 type arguments, but got 1.
@woocommerce/data:turbo:build: src/navigation/with-navigation-hydration.tsx(23,26): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/data:turbo:build: src/navigation/with-navigation-hydration.tsx(31,8): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/navigation/with-navigation-hydration.tsx(32,8): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/navigation/with-navigation-hydration.tsx(47,12): error TS2604: JSX element type 'OriginalComponent' does not have any construct or call signatures.
@woocommerce/data:turbo:build: src/onboarding/index.ts(21,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/onboarding/with-onboarding-hydration.tsx(19,37): error TS2558: Expected 2 type arguments, but got 1.
@woocommerce/data:turbo:build: src/onboarding/with-onboarding-hydration.tsx(23,61): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/data:turbo:build: src/onboarding/with-onboarding-hydration.tsx(28,25): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/onboarding/with-onboarding-hydration.tsx(29,35): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/onboarding/with-onboarding-hydration.tsx(71,12): error TS2604: JSX element type 'OriginalComponent' does not have any construct or call signatures.
@woocommerce/data:turbo:build: src/options/controls.ts(5,10): error TS2305: Module '"@wordpress/data"' has no exported member 'Action'.
@woocommerce/data:turbo:build: src/options/index.ts(21,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/options/with-options-hydration.tsx(47,30): error TS2558: Expected 2 type arguments, but got 1.
@woocommerce/data:turbo:build: src/options/with-options-hydration.tsx(51,12): error TS2604: JSX element type 'OriginalComponent' does not have any construct or call signatures.
@woocommerce/data:turbo:build: src/orders/index.ts(18,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/orders/resolvers.ts(28,19): error TS2344: Type 'OrdersQuery<OrdersQueryStatus>' does not satisfy the constraint 'BaseQueryParams<string>'.
@woocommerce/data:turbo:build:   Types of property '_fields' are incompatible.
@woocommerce/data:turbo:build:     Type '(string | number | symbol)[]' is not assignable to type 'string[]'.
@woocommerce/data:turbo:build: src/orders/resolvers.ts(48,41): error TS2344: Type 'OrdersQuery<OrdersQueryStatus>' does not satisfy the constraint 'BaseQueryParams<string>'.
@woocommerce/data:turbo:build: src/orders/types.ts(4,10): error TS2305: Module '"@wordpress/core-data/build-types"' has no exported member 'Schema'.
@woocommerce/data:turbo:build: src/plugins/index.ts(20,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/plugins/with-plugins-hydration.tsx(23,30): error TS2558: Expected 2 type arguments, but got 1.
@woocommerce/data:turbo:build: src/plugins/with-plugins-hydration.tsx(72,12): error TS2604: JSX element type 'OriginalComponent' does not have any construct or call signatures.
@woocommerce/data:turbo:build: src/product-form/index.ts(21,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/products/index.ts(18,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/products/resolvers.ts(46,19): error TS2344: Type 'ProductQuery<ProductStatus, ProductType>' does not satisfy the constraint 'BaseQueryParams<string>'.
@woocommerce/data:turbo:build:   Types of property '_fields' are incompatible.
@woocommerce/data:turbo:build:     Type '(string | number | symbol)[]' is not assignable to type 'string[]'.
@woocommerce/data:turbo:build:       Type 'string | number | symbol' is not assignable to type 'string'.
@woocommerce/data:turbo:build:         Type 'number' is not assignable to type 'string'.
@woocommerce/data:turbo:build: src/products/resolvers.ts(88,41): error TS2344: Type 'ProductQuery<ProductStatus, ProductType>' does not satisfy the constraint 'BaseQueryParams<string>'.
@woocommerce/data:turbo:build: src/products/types.ts(4,10): error TS2305: Module '"@wordpress/core-data/build-types"' has no exported member 'Schema'.
@woocommerce/data:turbo:build: src/reports/utils.ts(318,18): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/reports/utils.ts(337,20): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/reviews/index.ts(24,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/settings/index.ts(24,1): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/data:turbo:build: src/settings/use-settings.ts(23,25): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/settings/use-settings.ts(28,7): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/settings/use-settings.ts(30,20): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/settings/use-settings.ts(31,15): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/settings/with-settings-hydration.tsx(15,30): error TS2558: Expected 2 type arguments, but got 1.
@woocommerce/data:turbo:build: src/settings/with-settings-hydration.tsx(30,25): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/settings/with-settings-hydration.tsx(33,35): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/settings/with-settings-hydration.tsx(60,12): error TS2604: JSX element type 'OriginalComponent' does not have any construct or call signatures.
@woocommerce/data:turbo:build: src/user/types.ts(37,25): error TS2694: Namespace '"/Users/samseay/Code/automattic/woo/woocommerce/node_modules/.pnpm/@wordpress+core-data@6.3.3_react@17.0.2/node_modules/@wordpress/core-data/build-types/index"' has no exported member 'Schema'.
@woocommerce/data:turbo:build: src/user/types.ts(37,60): error TS2694: Namespace '"/Users/samseay/Code/automattic/woo/woocommerce/node_modules/.pnpm/@wordpress+core-data@6.3.3_react@17.0.2/node_modules/@wordpress/core-data/build-types/index"' has no exported member 'Schema'.
@woocommerce/data:turbo:build: src/user/types.ts(39,9): error TS2694: Namespace '"/Users/samseay/Code/automattic/woo/woocommerce/node_modules/.pnpm/@wordpress+core-data@6.3.3_react@17.0.2/node_modules/@wordpress/core-data/build-types/index"' has no exported member 'Schema'.
@woocommerce/data:turbo:build: src/user/types.ts(40,9): error TS2694: Namespace '"/Users/samseay/Code/automattic/woo/woocommerce/node_modules/.pnpm/@wordpress+core-data@6.3.3_react@17.0.2/node_modules/@wordpress/core-data/build-types/index"' has no exported member 'Schema'.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(121,2): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(124,19): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(129,4): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(131,4): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(133,4): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(139,5): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(140,7): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/use-user-preferences.ts(141,10): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/use-user.ts(18,19): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/data:turbo:build: src/user/use-user.ts(20,3): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/use-user.ts(26,5): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/use-user.ts(27,7): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/use-user.ts(29,10): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(20,30): error TS2558: Expected 2 type arguments, but got 1.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(24,26): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(28,5): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(32,8): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(33,8): error TS2349: This expression is not callable.
@woocommerce/data:turbo:build:   Type 'never' has no call signatures.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(38,5): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(40,5): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/data:turbo:build: src/user/with-current-user-hydration.tsx(51,12): error TS2604: JSX element type 'OriginalComponent' does not have any construct or call signatures.
```

**Comment by samueljseay** [comment]
Error report for `packages/js/components`:

```
 src/dynamic-form/field-types/field-select.tsx(37,4): error TS2604: JSX element type 'SelectControl' does not have any construct or call signatures.
@woocommerce/components:turbo:build: src/plugins/index.tsx(54,27): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/components:turbo:build: src/plugins/index.tsx(60,5): error TS2349: This expression is not callable.
@woocommerce/components:turbo:build:   Type 'never' has no call signatures.
@woocommerce/components:turbo:build: src/plugins/index.tsx(61,5): error TS2349: This expression is not callable.
@woocommerce/components:turbo:build:   Type 'never' has no call signatures.
@woocommerce/components:turbo:build: src/plugins/index.tsx(62,19): error TS2349: This expression is not callable.
@woocommerce/components:turbo:build:   Type 'never' has no call signatures.
@woocommerce/components:turbo:build: src/plugins/index.tsx(63,22): error TS2349: This expression is not callable.
@woocommerce/components:turbo:build:   Type 'never' has no call signatures.
@woocommerce/components:turbo:build: src/plugins/index.tsx(95,14): error TS7006: Parameter 'response' implicitly has an 'any' type.
@woocommerce/components:turbo:build: src/plugins/index.tsx(98,15): error TS7006: Parameter 'response' implicitly has an 'any' type.
@woocommerce/components:turbo:build: src/product-fields/store/index.ts(15,2): error TS2578: Unused '@ts-expect-error' directive.
@woocommerce/components:turbo:build: src/rich-text-editor/editor-writing-flow.tsx(37,49): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/components:turbo:build: src/search/index.tsx(320,6): error TS2604: JSX element type 'SelectControl' does not have any construct or call signatures.
```

**Comment by samueljseay** [comment]
Error report for `packages/js/customer-effort-score`:

```
@woocommerce/customer-effort-score:turbo:build: src/components/customer-effort-score-modal-container/index.tsx(25,6): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/customer-effort-score:turbo:build: src/components/customer-effort-score-modal-container/index.tsx(31,6): error TS2349: This expression is not callable.
@woocommerce/customer-effort-score:turbo:build:   Type 'never' has no call signatures.
@woocommerce/customer-effort-score:turbo:build: src/components/customer-effort-score-modal-container/index.tsx(35,6): error TS2349: This expression is not callable.
@woocommerce/customer-effort-score:turbo:build:   Type 'never' has no call signatures.
@woocommerce/customer-effort-score:turbo:build: src/components/customer-effort-score-modal-container/index.tsx(41,25): error TS2349: This expression is not callable.
@woocommerce/customer-effort-score:turbo:build:   Type 'never' has no call signatures.
@woocommerce/customer-effort-score:turbo:build: src/hooks/use-customer-effort-score-modal/index.ts(18,44): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/customer-effort-score:turbo:build: src/hooks/use-customer-effort-score-modal/index.ts(23,6): error TS2349: This expression is not callable.
@woocommerce/customer-effort-score:turbo:build:   Type 'never' has no call signatures.
@woocommerce/customer-effort-score:turbo:build: src/hooks/use-customer-effort-score-modal/index.ts(25,23): error TS2349: This expression is not callable.
@woocommerce/customer-effort-score:turbo:build:   Type 'never' has no call signatures.
@woocommerce/customer-effort-score:turbo:build: src/utils/customer-effort-score-exit-page.ts(25,11): error TS7006: Parameter 'trackingOption' implicitly has an 'any' type.
@woocommerce/customer-effort-score:turbo:build:  ELIFECYCLE  Command failed with exit code 2.
@woocommerce/customer-effort-score:turbo:build:  ELIFECYCLE  Command failed with exit code 1.
@woocommerce/customer-effort-score:turbo:build: ERROR: command finished with error: command (/Users/samseay/Code/automattic/woo/woocommerce/packages/js/customer-effort-score) pnpm run turbo:build exited (1)
```

**Comment by samueljseay** [comment]
Error report for `packages/js/product-editor`:

```
@woocommerce/product-editor:turbo:build: src/blocks/conditional/edit.tsx(26,27): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/blocks/conditional/edit.tsx(27,20): error TS2339: Property 'getEditedEntityRecord' does not exist on type 'never'.
@woocommerce/product-editor:turbo:build: src/blocks/name/edit.tsx(49,27): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/blocks/name/edit.tsx(50,20): error TS2339: Property 'getEditedEntityRecord' does not exist on type 'never'.
@woocommerce/product-editor:turbo:build: src/blocks/name/edit.tsx(64,47): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/blocks/shipping-class/edit.tsx(123,5): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/blocks/shipping-class/edit.tsx(199,7): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/product-editor:turbo:build: src/blocks/shipping-class/edit.tsx(202,17): error TS7006: Parameter 'value' implicitly has an 'any' type.
@woocommerce/product-editor:turbo:build: src/blocks/shipping-dimensions/edit.tsx(63,19): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/blocks/shipping-dimensions/edit.tsx(64,16): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/attribute-term-input-field/attribute-term-input-field.tsx(67,11): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/product-editor:turbo:build: src/components/attribute-term-input-field/attribute-term-input-field.tsx(75,8): error TS7006: Parameter 'attributeTerms' implicitly has an 'any' type.
@woocommerce/product-editor:turbo:build: src/components/attribute-term-input-field/attribute-term-input-field.tsx(80,8): error TS7006: Parameter 'error' implicitly has an 'any' type.
@woocommerce/product-editor:turbo:build: src/components/block-icon/block-icon.tsx(20,23): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/block-icon/block-icon.tsx(28,22): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/block-icon/block-icon.tsx(29,18): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/content-preview/content-preview.tsx(62,31): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/components/details-categories-field/use-category-search.ts(170,44): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/components/details-name-field/details-name-field.tsx(34,47): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/components/header/header.tsx(35,11): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-preview/use-preview.tsx(48,21): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-preview/use-preview.tsx(56,15): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-preview/use-preview.tsx(116,35): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-publish/use-publish.tsx(42,15): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-publish/use-publish.tsx(75,35): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-save-draft/use-save-draft.tsx(41,21): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-save-draft/use-save-draft.tsx(49,15): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/header/hooks/use-save-draft/use-save-draft.tsx(84,35): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/product-editor:turbo:build: src/components/iframe-editor/secondary-sidebar/inserter-sidebar.tsx(35,27): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/components/product-mvp-feedback-modal-container/product-mvp-feedback-modal-container.tsx(22,39): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/components/product-mvp-feedback-modal-container/product-mvp-feedback-modal-container.tsx(25,30): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/components/tabs/tabs.tsx(39,27): error TS2554: Expected 2 arguments, but got 1.
@woocommerce/product-editor:turbo:build: src/components/tabs/tabs.tsx(40,20): error TS2339: Property 'getEditedEntityRecord' does not exist on type 'never'.
@woocommerce/product-editor:turbo:build: src/hooks/use-confirm-unsaved-product-changes.ts(26,15): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/hooks/use-feedback-bar/use-feedback-bar.ts(19,33): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/hooks/use-feedback-bar/use-feedback-bar.ts(23,23): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
@woocommerce/product-editor:turbo:build: src/hooks/use-product-attributes.ts(51,11): error TS2347: Untyped function calls may not accept type arguments.
@woocommerce/product-editor:turbo:build: src/hooks/use-product-attributes.ts(59,8): error TS7006: Parameter 'attributeTerms' implicitly has an 'any' type.
@woocommerce/product-editor:turbo:build: src/hooks/use-product-attributes.ts(62,8): error TS7006: Parameter 'error' implicitly has an 'any' type.
@woocommerce/product-editor:turbo:build: src/hooks/use-product-edits.ts(31,19): error TS2349: This expression is not callable.
@woocommerce/product-editor:turbo:build:   Type 'never' has no call signatures.
```

**Comment by timmy5685** [comment]
Hi @samueljseay - thanks for tracking all of this. We're going through some older issues to sort out what's still relevant, and this one's long since resolved.

The repo is now on WP 6.9+ (tested to 7.0), with the @wordpress packages many majors past wp-6.2. Since the TS build gates CI (tsc --build), every upgrade between 6.2 and 6.9 required a clean TypeScript build — so all the errors tracked here, including the last unchecked woocommerce-admin item, were necessarily fixed along the way. That path is gone now anyway, since woocommerce-admin was merged into plugins/woocommerce/client/admin.

The draft PR (#39117) was closed without merging, so the wp-6.2 bump ended up shipping through other/incremental work rather than that branch — but it shipped.

Going to close this as resolved but feel free to open if you think this requires further attention.
